Packages/com.chisel.editor/Chisel/Editor/Components/Generators/CSGBox/Editor/CSGBoxGeneratorMode.cs

```diff
@@ -76,19 +78,20 @@ public void OnSceneGUI(SceneView sceneView, Rect dragArea)
                     CSGEditModeManager.EditMode = CSGEditMode.ShapeEdit;
                     break;
                 }
+
                 case BoxExtrusionState.Cancel:
                 {
-                    if (box)
-                        UnityEngine.Object.DestroyImmediate(box.gameObject);
                     Reset();
+                    Undo.RevertAllInCurrentGroup();
+                    EditorGUIUtility.ExitGUI();
```

Comment:
By using undo, the created box and, the optionally created model would both be removed.
EditorGUIUtility.ExitGUI prevents an error to occur in unity scenevieuw code